server/package/extractor/doc-generator.tsx

```diff
@@ -303,6 +304,139 @@ function setupApp(app: td.Application) {
   app.renderer.defineTheme("tsdocs", CustomTheme);
 }
 
+async function getMatchingTag(userName = "", repoName = "", version = "") {
+  const allTags = await getTagsData(userName, repoName);
+  const tag = allTags?.find(r => new RegExp(`((${repoName})|(v))[\\\-\\\_\\\@]{0,1}(${version})`, "i").test(r.ref))
+  return tag?.ref?.replace("refs/tags/", "");
+}
+
+const getReplacementText = (
+  key: string,
+  userName: string,
+  repoName: string,
+  tag: string,
+  packageRoot: string,
+  regexLeftSection = "",
+  regexRightSection = "",
+  type = "blob",
+) => {
+  if (
+    new RegExp(
+      `^[${regexLeftSection}](((\\\.\\\/){0,1}|(\\\.\\\.\\\/)+|([\\\w-_]+\\\/)+)([\\\w-_]+\\\/)*[\\\w-_]+\\\.[\\\w]{1,5})[${regexRightSection}]$`,
+    ).test(key)
+  ) {
+    let replacement = key.replace(
+      new RegExp(`(^[${regexLeftSection}]|[${regexRightSection}]$)`, "g"),
+      "",
+    );
+    if (replacement.startsWith("./")) {
+      replacement = replacement.replace(
+        /^\.\//,
+        `https://github.com/${userName}/${repoName}/${type}/${tag}/${packageRoot}${
+          packageRoot ? "/" : ""
+        }`,
+      );
+      return replacement;
+    }
+    if (/^(\.\.\/)+/.test(replacement)) {
+      const [relativePath] = Array.from(/^(\.\.\/)+/.exec(replacement));
+      let root = path.normalize(`${packageRoot}/${relativePath}`);
+      root = root === ".\\" ? "" : root;
+      replacement = replacement.replace(
+        /^(\.\.\/)+/,
+        `https://github.com/${userName}/${repoName}/${type}/${tag}/${root}`,
+      );
+      return replacement;
+    }
+    if (/^([\w-_]+\/)+/.test(replacement)) {
+      replacement = `https://github.com/${userName}/${repoName}/${type}/${tag}/${replacement}`;
+      return replacement;
+    }
+
+    return null;
+  }
+  return null;
+};
+
+async function updateReadmeRelativeLinks(
+  obj: td.Models.ProjectReflection,
+  packageName = "",
+  packageVersion = "",
+) {
+  try {
+    const relativeUrlRegex = new RegExp(
+      /(["|'|\()](((\.\/){0,1}|(\.\.\/)+|([\w-_]+\/)+)([\w-_]+\/)*[\w-_]+\.[\w]{1,5})["|'|\)])/g,
+    );
```

Comment:
The regexes here seem pretty unreadable — we could use an appropriate parser (for html, say) instead, or split these up into multiple simpler replaces

Reply:
Didn't find the need of a parser for something that can be easily done with a regex. But, yes I can make the regex more readable.

Reply:
Made the regex a little more verbose